pss/pss_test.go

```diff
@@ -47,29 +48,66 @@ func TestTrojanChunkRetrieval(t *testing.T) {
 	payload := []byte("RECOVERY CHUNK")
 	topic := trojan.NewTopic("RECOVERY")
 
-	var ch chunk.Chunk
+	var monitor *Monitor
 
 	pss := NewPss(localStore)
 
 	// call Send to store trojan chunk in localstore
-	if ch, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+	if monitor, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
 		t.Fatal(err)
 	}
 
 	// verify store, that trojan chunk has been stored correctly
 	var storedChunk chunk.Chunk
-	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, ch.Address()); err != nil {
+	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, monitor.chunk.Address()); err != nil {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(ch, storedChunk) {
+	if !reflect.DeepEqual(monitor.chunk, storedChunk) {
 		t.Fatalf("store chunk does not match sent chunk")
 	}
 
 	// check if pinning makes a difference
 
 }
 
+func TestPssMonitor(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+
+	localStore := newMockLocalStore(t)
+
+	testTargets := [][]byte{
+		{57, 120},
+		{209, 156},
+		{156, 38},
+		{89, 19},
+		{22, 129}}
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY")
+
+	var monitor *Monitor
+
+	pss := NewPss(localStore)
+
+	// call Send to store trojan chunk in localstore
+	if monitor, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	for state := range monitor.states {
```

Comment:
shouldn't this should go inside a monitor func? so that the logging happens when the state is updated

Reply:
well, I suppose it makes sense.
You still have to explicitly call it in the test, or leave it async and call it on update of status as you mentioned before.